002-template.md

```diff
@@ -0,0 +1,107 @@
+- Start Date: 2021-07-23
+- RFC PR: (leave this empty)
+
+# Summary
+
+The ADF have a huge growing, the ADF is a communication way between Dealer CRM, the ADF can has any information, using xml tags. Once Company can has a template for share the lead information, for example Demo Dealer 1 can share information of the lead, but Dealer Demo 2 can share information about lead, vehicle and vendor.
+
+Both company must has a template for share the information. This template can be storage in the json params of ADF rules. We can set a default Rule with the default template , but if the company need a custom template must be duplicate the default rule and modify the new rule and set the custom template in the params.
+
+For use ADF must use [Workflows](https://github.com/bakaphp/kanvas-packages/tree/0.3/src/WorkflowsRules)
+
+# Example
+
+## Add Rules to Entity
+
+```
+<?php 
+
+declare(strict_types = 1);
+use Kanvas\Packages\WorkflowsRules\Contracts\Traits\RulesTrait;
+
+class Users extends BaseModel 
+{
+    use RulesTrait;
+}
+
+?>
+```
+
+## In Database
+You must add new email templates called ADF, this template must use the volt syntax
+<br>Example
+```
+<firstname>{{entity.firstname}}</firstname>
+<lastname>{{entity.lastname}}</lastname>
+
+```
+## The entity ADF
+
+This entity is a collection with the data of lead, and actual messages or feed. When this rule is fire, the Message entity sent the Lead as first argument and the actual feed as second arguments, after the workflow take the data and create a new array 
+``` 
+[
+    'lead' => $arg[0]->toDocuments(),
```

Comment:
@FredPeal we would need a trait for leads or enty entity AllowADF ,

So you can do $leads->toADF();